core/src/main/java/org/apache/brooklyn/core/entity/EntityAsserts.java

```diff
@@ -0,0 +1,192 @@
+    }
+
+    public static <T extends Entity> void assertPredicateEventuallyTrue(Map<?,?> flags, final T entity, final Predicate<? super T> predicate) {
+        Asserts.succeedsEventually((Map)flags, new Runnable() {
+            @Override public void run() {
+                Asserts.assertTrue(predicate.apply(entity), "predicate unsatisfied");
+            }});
+    }
+
+    public static <T> void assertAttributeEqualsContinually(final Entity entity, final AttributeSensor<T> attribute, final T expected) {
+        assertAttributeEqualsContinually(Maps.newLinkedHashMap(), entity, attribute, expected);
+    }
+
+    public static <T> void assertAttributeEqualsContinually(Map<?,?> flags, final Entity entity, final AttributeSensor<T> attribute, final T expected) {
+        Asserts.succeedsContinually(flags, new Runnable() {
+            @Override public void run() {
+                assertAttributeEquals(entity, attribute, expected);
+            }});
+    }
+
+    public static void assertGroupSizeEqualsEventually(final Group group, int expected) {
+        assertGroupSizeEqualsEventually(ImmutableMap.of(), group, expected);
+    }
+
+    public static void assertGroupSizeEqualsEventually(Map<?,?> flags, final Group group, final int expected) {
+        Asserts.succeedsEventually((Map)flags, new Runnable() {
+            @Override public void run() {
+                Collection<Entity> members = group.getMembers();
+                assertEquals(members.size(), expected, "members=" + members);
+            }});
+    }
+
+
+    /**
+     * Asserts that the entity's value for this attribute changes, by registering a subscription and checking the value.
+     *
+     * @param entity The entity whose attribute will be checked.
+     * @param attribute The attribute to check on the entity.
+     *
+     * @throws AssertionError if the assertion fails.
+     */
+    public static void assertAttributeChangesEventually(final Entity entity, final AttributeSensor<?> attribute) {
+        final Object origValue = entity.getAttribute(attribute);
+        final AtomicBoolean changed = new AtomicBoolean();
+        SubscriptionHandle handle = entity.subscriptions().subscribe(entity, attribute, new SensorEventListener<Object>() {
+            @Override public void onEvent(SensorEvent<Object> event) {
+                if (!Objects.equal(origValue, event.getValue())) {
+                    changed.set(true);
+                }
+            }});
+        try {
+            Asserts.succeedsEventually(new Runnable() {
+                @Override public void run() {
+                    Asserts.assertTrue(changed.get(), entity + " -> " + attribute + " not changed");
+                }});
+        } finally {
+            entity.subscriptions().unsubscribe(entity, handle);
+        }
+    }
+
+
+    @Beta @SafeVarargs
+    public static <T> void assertAttributeNever(final Entity entity, final AttributeSensor<T> attribute, T... disallowed) {
```

Comment:
can we call this `assertAttributeContinuallyNotEqualTo`?

add javadoc noting that this loops checking the value, it does not currently set up a subscription so a disallowed value might come up very quickly and be missed.  (we could fix this if needed.)

cleaner would be for this to call the next method, `assertAttribute...(Map, ...)`, with `null` flags, or perhaps to introduce an `assertAttributeContinually(..., Predicate)` and build up a predicate of `not(or(equalTo(disallowed[0]), equalTo(disallowed[1]), ...))`